topic-operator/src/main/java/io/strimzi/operator/topic/cruisecontrol/CruiseControlClient.java

```diff
@@ -88,6 +65,29 @@ static CruiseControlClient create(String serverHostname,
      * @return The error message.
      */
     Optional<String> errorMessage(HttpResponse<String> response);
+
+    /**
+     * Client configuration.
+     *
+     * @param serverHostname Server hostname.
+     * @param serverPort Server port.
+     * @param rackEnabled Whether rack awareness is enabled.
+     * @param sslEnabled Whether SSL is enabled.
+     * @param sslCertificate SSL certificate.
+     * @param authEnabled Whether authentication is enabled.
+     * @param authUsername Authentication username.
+     * @param authPassword Authentication password.
+     */
+    record Config(
```

Comment:
I do think it's useful to encapsulate the configuration so it can be passed around a single variable, however we still have to initialise this with 8 parameters in the specific order. Would it be better to either go the whole way and make this a full class which is Buildable to make it nicer to initialise it, or if sticking with the record at least include the validation here as a custom constructor? That way we are getting added value from the new object.

Reply:
Good point. I am fine with both. We use the builder pattern across the project in other places as well (i.e. broker configuration).

Reply:
I'm not suggesting we follow this pattern everywhere. What I meant is I can understand why this change has been suggested, however if we are going to add an additional object (whether record or class) I think in order for it to be worth the overhead it should be more than just a holder for some parameters, since as you said it's only called in a single place.

Reply:
Thanks for the review. Appreciated.

As you suggested, I'm moving the input validation to the Config object constructor, as it is something I already explored in a previous commit, and I agree that it adds more value.

Reply:
@ppatierno I opted for the validation suggestion instead. Not against the builder, but it would be yet another object, and there would be not much need right now.